File: src/main/java/ListFlatMap.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * Project: KotlinBasic
 * Create By: Chen.F.X
 * DateTime: 2024-09-07 22:53
 **/
public class ListFlatMap {
    public static void main(String[] args) {
        List<Integer> list = new ArrayList<>(Arrays.asList(1, 2, 3));

        list.stream().flatMap(e -> {
            ArrayList<Integer> integers = new ArrayList<>(e);
            for (int i = 0; i < e; i++) {
                integers.add(i);
            }
            return integers.stream();
        }).forEach(it -> {
            System.out.println("result：" + it);
        });

    }
}

File: src/main/kotlin/com/chapter05/ListFlatMap.kt
package com.chapter05

import com.chapter04.times

/**
 * Project: KotlinBasic
 * Create By: Chen.F.X
 * DateTime: 2024-09-07 23:19
 *
 * flatMap 函数
 **/
fun main() {
    val list = listOf(1, 2, 3)

    list.flatMap {
        0 until it
    }.forEach {
        println("result: $it")
    }

    println("=" * 30)
    list.asSequence()
        .flatMap {
            (0 until it).asSequence()
        }
        .joinToString()
        .let(::println)
}